blog/2024-06-04-how-to-choose-an-embedding-model/index.mdx

```diff
@@ -0,0 +1,187 @@
+| 3 | Donut | Indulge in these sweet, fried delights. Glazed, powdered, or filled with decadent creams and fruit jams, each bite is a delightful burst of flavor and nostalgia. |
+| 4 | Muffin | Moist and tender, these muffins are bursting with flavor. Whether you prefer classic blueberry, indulgent chocolate chip, or hearty bran, there's a muffin for every craving. |
+| 5 | Croissant | Buttery layers of flaky pastry are folded to perfection, creating a delicate and irresistible treat. Enjoy them plain, stuffed with savory fillings, or paired with your favorite coffee. |
+| 6 | Scone | Crumbly yet tender, these scones are the epitome of comfort. Enjoy them plain or studded with fruits, nuts, or chocolate chips, accompanied by a dollop of clotted cream and jam. |
+| 7 | Pretzel | Crunchy on the outside, soft on the inside, these pretzels are a salty snack lover's dream. Enjoy them twisted into traditional shapes or dipped in sweet or savory toppings for a delightful twist. |
+| 8 | Sandwich | Freshly baked bread is the foundation for these hearty sandwiches. Pile on layers of meats, cheeses, crisp vegetables, and flavorful spreads for a satisfying meal on the go. |
+
+### Create embeddings
+
+Now, let’s generate and store the corresponding vector embeddings in our Weaviate vector database. For this, you will first connect to a Weaviate client in [embedded mode](https://weaviate.io/developers/weaviate/installation/embedded) for simplicity ([but you can instantiate a Weaviate client locally or use our managed cloud service](https://weaviate.io/developers/weaviate/installation)). 
+
+```python
+import weaviate
+
+# Connect to Weaviate client in embedded mode for simplicity
+client = weaviate.connect_to_embedded(
+	headers = {
+		"X-OpenAI-Api-Key": os.environ["OPENAI_API_KEY"]
+		}
+)
+```
+
+Then, you will configure a data collection called `"Pastries"` and configure the properties and vectorizer. This example uses OpenAI’s `text-embedding-3-small` embedding model to vectorize the data at import and query time automatically. 
+
+```python
+import weaviate.classes as wvc
+from weaviate.classes.config import Property, DataType
+
+# Create data collection
+pastries = client.collections.create(
+    name="Pastries",
+    vectorizer_config=wvc.config.Configure.Vectorizer.text2vec_openai(model='text-embedding-3-small'),
+    properties=[
+        Property(name="pastry_name", data_type=DataType.TEXT),
+        Property(name="pastry_description", data_type=DataType.TEXT),
+    ]
+)
+```
+
+Finally, you can import your data:
+
+```python
+# Import data objects
+pastry_objects = list()
+for _, row in df.iterrows():
+    properties = {
+        "pastry_name": row.pastry_name,
+        "pastry_description": row.pastry_description
+    }
+    pastry_objects.append(properties)
+
+pastries.data.insert_many(pastry_objects)
+```
+
+### Retrieve
+
+Now, you can run a few queries:
+
+```python
+response = pastries.query.near_text(
```

Comment:
Similarly - wondering if we should import `MetadataQuery` with `from weaviate.classes.query import MetadataQuery` for completeness. wdyt?